tests/errors/tests/errors.js

```diff
@@ -21,6 +68,41 @@ describe("errors", () => {
     }
   });
 
+  it("Emits a Hello error via err!", async () => {
+    try {
+      const tx = await program.rpc.testErr();
+      assert.ok(false);
+    } catch (err) {
+      const errMsg =
+        "This is an error message clients will automatically display";
+      assert.equal(err.toString(), errMsg);
+      assert.equal(err.msg, errMsg);
+      assert.equal(err.code, 6000);
+    }
+  });
+
+  it("Logs a ProgramError", async () => {
+    await withLogTest(async () => {
+      try {
+        const tx = await program.rpc.testProgramError();
+        assert.ok(false);
+      } catch (err) {
+        // No-op (withLogTest expects the callback to catch the initial tx error)
+      }
+    }, "Program log: ProgramError occurred. Error Code: InvalidAccountData. Error Number: 17179869184. Error Message: An account's data contents was invalid.");
```

Comment:
dumb q: Is `17179869184` correct? Seems quite large?

Reply:
yea I was surprised by this as well but it matches the docs inside the ProgramError source. look for a macro called sth like "builtin"

Reply:
wild